test/solver.jl

```diff
@@ -142,8 +187,42 @@ k = 1
         h_U            = collect(d_U)
         h_Vt           = collect(d_Vt)
         svda           = svdfact(A,thin=false)
-        # @test h_U ≈ svda[:U]
-        # @test h_S ≈ svdvals(A)
-        # @test h_Vt ≈ svda[:Vt]
+        @test h_U ≈ svda[:U]
+        @test h_S ≈ svdvals(A)
+        @test h_Vt ≈ svda[:Vt]
     end
+
+    @testset "qr" begin
+        A              = rand(elty, m, n)
+        d_A            = CuArray(A)
+        d_q            = qrq!(d_A)
+        h_q            = collect(d_q)
+        q              = qrfact!(A)
+        @test h_q ≈ Array(q[:Q])
+        A              = rand(elty, n, m)
+        d_A            = CuArray(A)
+        d_q            = qrq!(d_A)
+        h_q            = collect(d_q)
+        q              = qrfact!(A)
+        @test h_q ≈ Array(q[:Q])
+        CuArrays.allowscalar(true)
```

Comment:
Why is this needed here? We shouldn't need this for any "supported" uses of CuArrays.

Reply:
The problem here is that the ```geqrf!``` function from cusolver returns a compact representation ```QR``` of the qr decomposition, with the ```R``` part being the upper triangular part of ```QR```. In making the high level implementation I just adjusted the qr implementation from julia base, where to return ```R``` they apply ```triu!``` which uses scalar indexing.